pyjob/sge.py

```diff
@@ -110,8 +127,12 @@ def _create_runscript(self):
             cmd = '-S {}'.format(self.shell)
             runscript.append(self.__class__.SCRIPT_DIRECTIVE + ' ' + cmd)
         if self.nprocesses:
-            cmd = '-pe {} {}'.format(self.environment, self.nprocesses)
-            runscript.append(self.__class__.SCRIPT_DIRECTIVE + ' ' + cmd)
+            if self.environment in self.available_environments:
+                cmd = '-pe {} {}'.format(self.environment, self.nprocesses)
+                runscript.append(self.__class__.SCRIPT_DIRECTIVE + ' ' + cmd)
+            else:
+                warnings.warn("Environment {} is not available, -pe card cannot be set.".format(self.environment))
+                warnings.warn("List of available environments: [ {} ]".format(", ".join(self.available_environments)))
```

Comment:
I *think* this should probably be a `logging.warning` call not a `warnings.warn`. Reason is that `warnings` would imply some API change, but `environment` is user-defined so bad / incompatible input.